codec-http/src/main/java/io/netty5/handler/codec/http/headers/HeaderUtils.java

```diff
@@ -63,6 +64,17 @@ public final class HeaderUtils {
     public static final BiFunction<? super CharSequence, ? super CharSequence, CharSequence> DEFAULT_HEADER_FILTER =
             (k, v) -> "<filtered>";
 
+    /**
+     * Whether cookie parsing should be strictly spec compliant with
+     * <a href="https://www.rfc-editor.org/rfc/rfc6265">RFC6265</a> ({@code true}), or allow some deviations that are
+     * commonly observed in practice and allowed by the obsolete
+     * <a href="https://www.rfc-editor.org/rfc/rfc2965">RFC2965</a>/
+     * <a href="https://www.rfc-editor.org/rfc/rfc2109">RFC2109</a> ({@code false}, the default).
+     */
+    // not final for testing
+    private static boolean cookieParsingStrictRfc6265 = SystemPropertyUtil.getBoolean(
```

Comment:
I get why we have this "non-final" but I wonder if this will cause the JIT not optimise correctly as it cant be sure this field is not changed during runtime.

Reply:
@chrisvest thoughts ?

Reply:
I don't see a good way to test that code path if this is `final`, but if it's on a shared cache line then the load instructions should be fast and the branch predictors should see through it. Compared to the `charAt` call that follows, I don't think loading this field will have much overhead.